internal/controllers/machine/machine_controller.go

```diff
@@ -687,6 +685,13 @@ func (r *Reconciler) shouldAdopt(m *clusterv1.Machine) bool {
 }
 
 func (r *Reconciler) watchClusterNodes(ctx context.Context, cluster *clusterv1.Cluster) error {
+	log := ctrl.LoggerFrom(ctx)
+
+	if cluster.GetConditions() == nil || conditions.IsFalse(cluster, clusterv1.ControlPlaneInitializedCondition) {
+		log.V(5).Info("Skipping node watching setup because control plane is not initialized", "cluster", cluster.Name)
```

Comment:
```suggestion
		log.V(5).Info("Skipping node watching setup because control plane is not initialized")
```
nit: the Cluster k/v pair is already added in `Reconcile` (on main)